src/cennznut/v0/mod.rs

```diff
@@ -68,20 +73,33 @@ impl CENNZnutV0 {
 
 impl Encode for CENNZnutV0 {
     fn encode_to<T: Output>(&self, buf: &mut T) {
-        buf.write(&[0, 0]);
-
-        #[allow(clippy::cast_possible_truncation)]
-        let module_count = (self.modules.len() as u8).swap_bits();
-        buf.push_byte(module_count);
+        if self.modules.is_empty() || self.modules.len() > MAX_MODULES {
+            return;
```

Comment:
Is there a way to avoid these silent failures? For example can we change the signature encode_to in the original trait so it returns Result?

Reply:
Sadly no, this is why I have decided to make it fail by returning nothing.
The `encode_to` interface is provided by the parity `scale` codec

Reply:
My preference would be returning a Result, but I think the reason it is this way was to allow nested encode calls